nix/redis-cluster_test.nix

```diff
@@ -1,16 +1,23 @@
-{ config, ... }: {
+{ pkgs, config, ... }: {
   services.redis-cluster."c1".enable = true;
-  testScript = ''
-    process_compose.wait_until(lambda procs:
-      # TODO: Check for 'is_ready' of `c1-cluster-create` instead of `c1-n1` (status of `c1-cluster-create` determines whether the hashslots are assigned). 
-      #       This should be easy after https://github.com/juspay/services-flake/issues/32
-      procs["c1-n1"]["status"] == "Running"
-    )
-    machine.succeed("${config.services.redis-cluster.c1.package}/bin/redis-cli -p 30001 ping | grep -q 'PONG'")
-    machine.succeed("${config.services.redis-cluster.c1.package}/bin/redis-cli -p 30002 ping | grep -q 'PONG'")
-    machine.succeed("${config.services.redis-cluster.c1.package}/bin/redis-cli -p 30003 ping | grep -q 'PONG'")
-    machine.succeed("${config.services.redis-cluster.c1.package}/bin/redis-cli -p 30004 ping | grep -q 'PONG'")
-    machine.succeed("${config.services.redis-cluster.c1.package}/bin/redis-cli -p 30005 ping | grep -q 'PONG'")
-    machine.succeed("${config.services.redis-cluster.c1.package}/bin/redis-cli -p 30006 ping | grep -q 'PONG'")
-  '';
+
+  settings.processes.test =
+    let
+      cfg = config.services.redis-cluster."c1";
+    in
+    {
+      command = pkgs.writeShellApplication {
+        text = ''
+          ${cfg.package}/bin/redis-cli -p 30001 ping | ${pkgs.gnugrep}/bin/grep -q "PONG"
+          ${cfg.package}/bin/redis-cli -p 30002 ping | ${pkgs.gnugrep}/bin/grep -q "PONG"
+          ${cfg.package}/bin/redis-cli -p 30003 ping | ${pkgs.gnugrep}/bin/grep -q "PONG"
+          ${cfg.package}/bin/redis-cli -p 30004 ping | ${pkgs.gnugrep}/bin/grep -q "PONG"
+          ${cfg.package}/bin/redis-cli -p 30005 ping | ${pkgs.gnugrep}/bin/grep -q "PONG"
+          ${cfg.package}/bin/redis-cli -p 30006 ping | ${pkgs.gnugrep}/bin/grep -q "PONG"
```

Comment:
You can avoid the `${foo}/bin` here by passing these packages to `runtimeInputs` argument of `writeShellApplication`.

Reply:
Same in postgres test.